lib/jekyll/drops/site_drop.rb

```diff
@@ -37,6 +37,16 @@ def collections
         @site_collections ||= @obj.collections.values.sort_by(&:label).map(&:to_liquid)
       end
 
+      # `{{ site.related_posts }}` is how posts can get posts related to
+      # them, either through LSI if it's enabled, or through the most
+      # recent posts.
+      # We should remove this in 4.0 and switch to `{{ post.related_posts }}`.
+      def related_posts
+        return nil unless @current_document.is_a?(Jekyll::Document)
```

Comment:
can we also return early if the current document is not a Post`..?
Like for a site with two posts and another *collection* with 1000 documents..?

Reply:
> can we also return early if the current document is not a `Post`..?

I don't think so. A site could be using `site.related_posts` on a Document to display the most recent blog posts.

Reply:
good point.. :+1: